phenex/phenotypes/computation_graph_phenotypes.py

```diff
@@ -96,6 +114,54 @@ def _execute(self, tables: Dict[str, Table]) -> PhenotypeTable:
 
         return joined_table
 
+    def _return_all_dates(self, table, date_columns):
+        """
+        If return date = all, we want to return all the dates on which phenotype criteria are fulfilled; this is a union of all the non-null dates in any leaf phenotype date columns.
+
+        Args:
+            table: The Ibis table object (e.g., joined_table) that contains all leaf phenotypes stacked horizontally
+            date_columns: List of base columns as ibis objects
+
+        Returns:
+            Ibis expression representing the UNION of all non null dates.
+        """
+        # get all the non-null dates for each date column
+        non_null_dates_by_date_col = []
+        for date_col in date_columns:
+            non_null_dates = (
+                table.filter(date_col.notnull())
+                    .mutate(EVENT_DATE=date_col)
+            )
+            non_null_dates_by_date_col.append(non_null_dates)
+
+        # do the union of all the non-null dates
+        all_dates = non_null_dates_by_date_col[0]
+        for non_null_dates in non_null_dates_by_date_col[1:]:
+            all_dates = all_dates.union(non_null_dates)
+        return all_dates
+
+    def _coalesce_all_date_columns(self, table):
+        """
+        Generate COALESCE strings for the GREATEST or LEAST function using Ibis.
```

Comment:
are we back to working with strings ... ?

Reply:
nope, it uses ibis but is doing coalesce... changed docstring

Reply:
Awesome